xbmc/addons/BinaryAddonCache.h

```diff
@@ -31,8 +31,9 @@ namespace ADDON {
 class CBinaryAddonCache : public Observer
 {
 public:
-  virtual ~CBinaryAddonCache();
+  virtual ~CBinaryAddonCache() { }
```

Comment:
Destructor is called from CApplication descructor, which is way too late. an Init() call should have a corresponding Deinit(). Init() is called right after CAddonMgr is inited, so Deinit() should be called right before CAddonMgr is Deinited